radix-common/src/crypto/bls12381/public_key.rs

```diff
@@ -26,8 +26,33 @@ impl Bls12381G1PublicKey {
         PublicKey::from_bytes(&self.0).map_err(|err| err.into())
     }
 
-    /// Aggregate multiple public keys into a single one
-    pub fn aggregate(public_keys: &[Bls12381G1PublicKey]) -> Result<Self, ParseBlsPublicKeyError> {
+    /// Aggregate multiple public keys into a single one.
+    /// This method does validate provided input keys.
```

Comment:
```suggestion
    /// This method validates provided input keys.
```
(although if we do the below and expose the boolean, this will need tweaking :)